backend/controllers/matchingController.js

```diff
@@ -0,0 +1,51 @@
+const express = require('express');
+const router = express.Router();
+const axios = require('axios');
+
+
+const mongoose = require('mongoose');
+
+const auth = require('../auth');
+const User = require("../models/User");
+
+const baseURL = process.env.BASE_URL || 'http://localhost:3000';
+
+// 1. Matching by role
+router.get('/lists', auth, async (req, res) => {
+    const userId = req.user.userId;
+    const userProfileUrl = `${baseURL}/users/userProfile/${userId}`;
+
+    // Use Axios to call the get user API to get the user's role
+    try {
+        const response = await axios.get(userProfileUrl, {
+            headers: {
+                Authorization: req.headers.authorization
+            }
+        });
+
+        const role = response.data.role;
+
+        const matchingRole = role === 'Mentee' ? 'Mentor' : role === 'Mentor' ? 'Mentee' : null;
```

Comment:
are we ever expecting a case where role will be null?  its either Mentee or Mentor, so we can get rid of the case if its nothing